docs/topics/multiplatform/multiplatform-android-layout.md

```diff
@@ -0,0 +1,107 @@
+
+## Move the `AndroidManifest.xml` file
+
+If you have the `AndroidManifest.xml` file in your project, move it to the new directory, following this pattern:
+
+| Previous source set layout                             | New source set layout                                 |
+|--------------------------------------------------------|-------------------------------------------------------|
+| src/{<b>Android</b>SourceSet.name}/AndroidManifest.xml | src/{<b>Kotlin</b>SourceSet.name}/AndroidManifest.xml |
+
+`{AndroidSourceSet.name}` maps to `{AndroidManifest.xml location}` as follows:
+
+|       | Previous source set layout    | New source set layout                       |
+|-------|-------------------------------|---------------------------------------------|
+| main  | src/main/AndroidManifest.xml  | src/<b>android</b>Main/AndroidManifest.xml  |
+| debug | src/debug/AndroidManifest.xml | src/<b>android</b>Debug/AndroidManifest.xml |
+
+## Check the relation between Android and common tests
+
+The new Android source set layout changes the relation between Android-instrumented tests (renamed to `androidInstrumentedTest` in the new layout)
+and common tests.
+
+Previously, there was a default `dependsOn` relation between `androidAndroidTest` and `commonTest`. In practice, it meant the following:
+
+* The code in `commonTest` was available in `androidAndroidTest`.
+* `expect` declarations in `commonTest` had to have corresponding `actual` implementations in `androidAndroidTest`.
+* Tests declared in `commonTest` were also running as Android instrumented tests.
+
+In the new Android source set layout, the `dependsOn` relation is not added by default. If you prefer the previous behavior,
+manually declare this relation in your `build.gradle.kts` file:
+
+```kotlin
+kotlin {
+// ...
+    sourceSets {
+        val commonTest by getting
+        val androidInstrumentedTest by getting {
+            dependsOn(commonTest)
+        }
+    }
+}
+```
+
+## Adjust the implementation of Android flavors
+
+Previously, the Kotlin Gradle plugin eagerly created source sets that correspond to Android source sets with `debug` and
+`release` build types or custom flavors like `demo` and `full`.
+It made them accessible by constructions like `val androidDebug by getting { ... }`.
+
+The new Android source set layout makes use of Android [`onVariants`](https://developer.android.com/reference/tools/gradle-api/8.0/com/android/build/api/variant/AndroidComponentsExtension#onVariants(com.android.build.api.variant.VariantSelector,kotlin.Function1))
+to create source sets. It makes such expressions invalid,
+leading to errors like `org.gradle.api.UnknownDomainObjectException: KotlinSourceSet with name 'androidDebug' not found`.
+
+To work around that, use the new `invokeWhenCreated()` API in your `build.gradle.kts` file:
+
+```kotlin
+kotlin {
+// ...
+    sourceSets.invokeWhenCreated("androidFreeDebug") {
```

Comment:
This API requires `@OptIn(ExperimentalKotlinGradlePluginApi::class)`. I don't know, do we usually show optIn's in samples? Maybe it will just create noise.

Reply:
Usually we do, added the opt-in annotation